src/nex_types.rs

```diff
@@ -0,0 +1,446 @@
+
+        let mut params = fields.split(";");
+
+        while let Some(param) = params.next() {
+            let mut split = param.split("=");
+
+            let name = split.next().unwrap_or_else(|| "");
+            let value: Option<String> = Some(split.next().unwrap_or_else(|| "").to_string());
+            match name {
+                "address" => station_url.address = value,
+                "port" => station_url.port = value,
+                "stream" => station_url.stream = value,
+                "sid" => station_url.sid = value,
+                "CID" => station_url.cid = value,
+                "PID" => station_url.pid = value,
+                "type" => station_url.transport_type = value,
+                "RVCID" => station_url.rvcid = value,
+                "natm" => station_url.natm = value,
+                "natf" => station_url.natf = value,
+                "upnp" => station_url.upnp = value,
+                "pmp" => station_url.pmp = value,
+                "probeinit" => station_url.probe_init = value,
+                "PRID" => station_url.prid = value,
+                _ => {}
+            }
+        }
+        station_url
+    }
+
+    pub fn encode_to_string(&self) -> String {
+        let mut fields = Vec::new();
+
+        if let Some(address) = &self.address {
+            fields.push(address.to_string());
+        }
+
+        if let Some(port) = &self.port {
+            fields.push(port.to_string());
+        }
+
+        if let Some(stream) = &self.stream {
+            fields.push(stream.to_string());
+        }
+
+        if let Some(sid) = &self.sid {
+            fields.push(sid.to_string());
+        }
+
+        if let Some(cid) = &self.cid {
+            fields.push(cid.to_string());
+        }
+
+        if let Some(pid) = &self.pid {
+            fields.push(pid.to_string());
+        }
+
+        if let Some(transport_type) = &self.transport_type {
+            fields.push(transport_type.to_string());
+        }
+
+        if let Some(rvcid) = &self.rvcid {
+            fields.push(rvcid.to_string());
+        }
+
+        if let Some(natm) = &self.natm {
+            fields.push(natm.to_string());
+        }
+
+        if let Some(natf) = &self.natf {
+            fields.push(natf.to_string());
+        }
+
+        if let Some(upnp) = &self.upnp {
+            fields.push(upnp.to_string());
+        }
+
+        if let Some(pmp) = &self.pmp {
+            fields.push(pmp.to_string());
+        }
+
+        if let Some(probe_init) = &self.probe_init {
+            fields.push(probe_init.to_string());
+        }
+
+        if let Some(prid) = &self.prid {
+            fields.push(prid.to_string());
+        }
+
+        let fields_string = fields.join(";");
+        let scheme = if let Some(scheme) = &self.scheme {
+            scheme.to_string()
+        } else {
+            "".to_string()
+        };
+
+        format!("{}:/{}", scheme, fields_string)
+    }
+}
+
+struct Result {
```

Comment:
What are thoughts on naming this to `ResultCode` to avoid confusion with rust's `Result`?

Reply:
Changed to a tuple struct as well for simplification